lib/db/mem.js

```diff
@@ -1,45 +1,61 @@
-var db = []
-
-function match(event, filter) {
-  var filterNames = Object.keys(filter)
-  for (var i = 0; i < filterNames.length; i++) {
-    var name = filterNames[i]
-    if (event[name] && event[name] === filter[name]) {
-      continue
-    }
-    else {
-      return false
+var jws = require('jws')
+
+function match(filter, str) {
+  try { // too lazy
+    var event = JSON.parse(jws.decode(str).payload)
+    var filterNames = Object.keys(filter)
+    for (var i = 0; i < filterNames.length; i++) {
+      var name = filterNames[i]
+      if (event[name] && event[name] === filter[name]) {
+        continue
+      }
+      else {
+        return false
+      }
     }
+    return true
+  } catch (e) {
+    return false
```

Comment:
You may be able to shorten that `for ()` loop up a bit (and avoid the `continue`) using something like `Array.some()`:

```js
function match(filter, str) {
  try { // too lazy
    var event = JSON.parse(jws.decode(str).payload)
    return Object.keys(filter).some(function (name) {
      return (event[name] && event[name] === filter[name])
    })
  } catch (e) {
    return false
  }
}
```